torchaudio/datasets/commonvoice.py

```diff
@@ -196,23 +203,12 @@ def __init__(self,
 
         self._path = os.path.join(root, folder_in_archive)
 
-        if not os.path.isdir(self._path):
-            if os.path.isfile(archive):
-                checksum = _CHECKSUMS.get(url, None)
-                if checksum:
-                    filepath = os.path.basename(url)
-                    with open(filepath, "rb") as file_obj:
-                        if not validate_file(file_obj, checksum, "sha256"):
-                            raise RuntimeError(
-                                f"The hash of {filepath} does not match. Delete the file manually and retry."
-                            )
+        if download:
+            if not os.path.isdir(self._path):
+                if not os.path.isfile(archive):
+                    checksum = _CHECKSUMS.get(url, None)
+                    download_url(url, root, hash_value=checksum)
```

Comment:
This will never run, so the file needs to be extracted by the user.

Reply:
Updated the error message to `extract`.